code/ecdsap256sha2/Hacl.Impl.LowLevel.fst

```diff
@@ -0,0 +1,828 @@
+module Hacl.Impl.LowLevel
+
+open FStar.HyperStack.All
+open FStar.HyperStack
+module ST = FStar.HyperStack.ST
+
+open Lib.IntTypes
+open Lib.Buffer
+
+open Hacl.Spec.P256.Definitions
+open Hacl.Spec.P256.Lemmas
+open Hacl.Spec.ECDSA
+
+open FStar.Math
+open FStar.Math.Lemmas
+open FStar.Mul
+
+open FStar.Tactics
+open FStar.Tactics.Canon 
+
+open Hacl.Spec.P256.Lemmas
+
+#reset-options "--z3rlimit 400"
+
+(* This code is not side channel resistant *)
```

Comment:
These comments are scary. 
Are any of the non constant-time functions used in code that needs to be constant-time?

Reply:
For clarity, the functions that break the `Lib.IntTypes` abstraction and are thus not side-channel resistant should be put in a separate module that is only locally opened in the places where it is used. All functions transitively using any of the functions in this module should be clearly documented as non constant-time.

Reply:
For instance, the top-level function `Hacl.Impl.ECDSA.ecdsa_p256_sha2_sign` transitively uses `Hacl.Impl.LowLevel.eq_u64_nCT`:

`Hacl.Impl.LowLevel.eq_u64_nCT -> Hacl.Impl.LowLevel.sub_borrow -> Hacl.Impl.LowLevel.sub4_il -> Hacl.Impl.ECDSA.MontgomeryMultiplication.reduction_prime_2prime_with_carry -> Hacl.Impl.ECDSA.MontgomeryMultiplication.montgomery_multiplication_ecdsa_module -> Hacl.Impl.ECDSA.P256SHA256.Signature.ecdsa_signature_step6 -> Hacl.Impl.ECDSA.P256SHA256.Signature.ecdsa_signature_core -> Hacl.Impl.ECDSA.P256SHA256.Signature.ecdsa_signature ->
Hacl.Impl.ECDSA.ecdsa_p256_sha2_sign`

Is this usage ok? If so, how can we make it clear that the timing doesn't depend on the secret key?

Reply:
Signing is now constant-time using intrinsics for `_addcarry_u64` and `_subborrow_u64`. Thanks @Frosne!
The remaining non-constant-time functions are documented in https://github.com/project-everest/hacl-star/blob/anilam_ci/code/ecdsap256sha2/README.md. Some are only used internally and the rest are non-constant-time on public values only (e.g. signature verification).